send.go

```diff
@@ -115,8 +117,14 @@ func (c *Connection) send(ctx context.Context, request *http.Request) (response
 		return
 	}
 
+	var token string
 	// Get the access token:
-	token, _, err := c.TokensContext(ctx)
+	operation := func() error {
+		token, _, err = c.TokensContext(ctx)
```

Comment:
Not a blocker, but do we want to log retries?

Reply:
Good idea. I added it.